src/apprt/gtk/Window.zig

```diff
@@ -588,6 +589,17 @@ fn gtkRealize(v: *c.GtkWindow, ud: ?*anyopaque) callconv(.C) bool {
     return true;
 }
 
+fn gtkWindowNotifyMaximized(
+    _: *c.GObject,
+    _: *c.GParamSpec,
+    ud: ?*anyopaque,
+) callconv(.C) void {
+    const self = userdataSelf(ud orelse return);
+    if (self.app.config.@"gtk-titlebar-hide-when-maximized") {
```

Comment:
I think there's potentially a bug here, but haven't verified:

1. Set `gtk-titlebar-hide-when-maximized`
2. Maximize
3. Unset and reload config
4. Unmaximize, header bar is now missing.

I think the fix is easy enough: unconditionally show the header bar when the window is unmaximized.